### PR TITLE
Prepare for major release 4.0.0

### PR DESCRIPTION
**Pull request by Ruchip16**
##### SUMMARY

changelog PR for major release 8.0.0

##### ISSUE TYPE

 - changelog PR

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -365,3 +365,18 @@ New Modules
...
+-------------
+
+- This release removes deprecated is_private() function in favour of netaddr is_global(). Please refer to the **Removed Features** section for details.
+
+Removed Features (previously deprecated)
+----------------------------------------
+
+- Remove deprecated is_private() function which is replaced with netaddr is_global() function.
+- Remove the IPAddress.is_private method.
+- Set netaddr minimum version required to 0.10.1
```
Comment (NilashishC) [inline CHANGELOG.rst:382]: Can you move this line to major changes?

Overall, from the release notes it isn't clear what user facing  features are being impacted with this major release. In fact, even the affected filter plugin isn't mentioned here. This only talks about code change, which is abstracted to the user anyway.

Comment (NilashishC) [reply CHANGELOG.rst:382]: I don't think you need a "Removed Features" section here. Rather, an explanation in "Major Changes" about _why_ we had to merge https://github.com/ansible-collections/ansible.utils/pull/338.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -365,3 +365,14 @@ New Modules
...
 - update_fact - Update currently set facts
 - validate - Validate data with provided criteria
+
+v4.0.0
+======
+
+Major Changes
+-------------
+
+- This release mainly fixes the breaking changes in the `netaddr` library.
```
Comment (NilashishC) [inline CHANGELOG.rst:375]:
```suggestion
- This release mainly addresses the breaking changes in the `netaddr` library.
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -365,3 +365,14 @@ New Modules
...
+- With the new release of `netaddr` 1.0.0, the `IPAddress.is_private()` method has been removed and instead, the `IPAddress.is_global()` method has been extended to support the same functionality. This change has been reflected in the `ipaddr` filter plugin.
+- Bumping `netaddr` to `>=0.10.1`, means that starting from this release, the minimum `netaddr` version this collection requires is `>=0.10.1`.
+- Removes compat files as we stopped testing against python 2.7.
```
Comment (NilashishC) [inline CHANGELOG.rst:378]: test related updates do not need to be part of changelog.

Comment (Ruchip16) [reply CHANGELOG.rst:378]: cool, removing them

**Review by NilashishC — APPROVED** [review]

**Pull request merged**